src/primitives/polyline/scanline_intersections.rs

```diff
@@ -89,103 +89,105 @@ impl<'a> ScanlineIntersections<'a> {
     }
 }
 
-// PERF: There may be some performance to be gained by merging adjacent line segments - this would
-// shrink the number of draw calls around polyline joints and any line segments that are directly
-// next to each other.
+/// This iterator loops through all scanline intersections for all segments. If two intersections
+/// are adjacent or overlapping, an accumulator line is extended. This repeats until the next
+/// intersection does not touch the current accumulator. At this point, the accumulated line
+/// segment is returned, and is reset to the next segment.
+///
+/// This process reduces the number of draw calls for adjacent scanlines, whilst preventing overdraw
+/// from overlapping scanline segments.
+///
+/// ```text
+/// # Adjacent - merge
+/// X---XY+++Y
+///     ⇓
+/// X--------X
+///
+/// # Overlapping - merge
+/// X---Y+++X+++Y
+///      ⇓
+/// X-----------X
+///
+/// # Separate - leave alone
+/// X---X Y---Y
+///      ⇓
+/// X---X Y---Y
```

Comment:
I was very briefly confused by X and Y, because I hadn't read the text and thought the lines were using a different orientation. IMO it's better to use other markers:
```suggestion
/// A---AB+++B
///     ⇓
/// A--------A
///
/// # Overlapping - merge
/// A---B+++A+++B
///      ⇓
/// A-----------A
///
/// # Separate - leave alone
/// A---A B---B
///      ⇓
/// A---A B---B
```